herald/src/main/java/io/heraldprox/herald/sensor/data/TextFile.java

```diff
@@ -22,28 +20,38 @@
 import java.util.ArrayList;
 import java.util.Collection;
 import java.util.List;
+import java.util.concurrent.ConcurrentLinkedQueue;
 import java.util.concurrent.Executors;
 import java.util.concurrent.ScheduledExecutorService;
 import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
 
 public class TextFile implements Resettable {
     private final SensorLogger logger = new ConcreteSensorLogger("Sensor", "Data.TextFile");
+    private final ScheduledExecutorService executorService = Executors.newScheduledThreadPool(1);
     @NonNull
     private final File file;
+    private final ConcurrentLinkedQueue<String> writeBuffer = new ConcurrentLinkedQueue<>();
+    private final AtomicLong writeBufferSize = new AtomicLong();
 
     public TextFile(@NonNull final File file) {
         this.file = file;
         final File folder = file.getParentFile();
-        if (!folder.exists()) {
+        if (folder != null && !folder.exists()) {
```

Comment:
May want to add two tests - one for a file within a folder that does not exist, another for a file within a folder within a folder that does not exist. Will also want to handle the situation where the folder exists, but create or write permissions to the file is not given, or the filesystem becomes full (E.g. during a long test).

Reply:
Updated checks to use two separate tests to provide better logging.